### PR TITLE
Allow notifications to be disabled

### PR DESCRIPTION
**Pull request by ouroboros8**

### CODE REVIEW

File: config/store_config.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ type StoreConfig struct {
 	StoreBackend  string `yaml:"storebackend"`  // storage backend (e.g. FS, K/V, ...)
 	SyncBackend   string `yaml:"syncbackend"`   // sync backend (e.g. GitCLI, GoGit, ...)
 	UseSymbols    bool   `yaml:"usesymbols"`    // always use symbols when generating passwords
+	Notify        bool   `yaml:"notify"`        // enable desktop notifications
```
Comment (dewey) [inline config/store_config.go]: Maybe it's just me but in a config option to just call it Notify (like the internal library) could be confusing. Maybe EnableNotifications, or just Notifications would be clearer to the outside?

Comment (ouroboros8) [reply config/store_config.go]: Agree! Let's go with Notifications, as it's a bit more concise.

File: utils/notify/notify_darwin.go [hunk 2]
```diff
@@ -3,13 +3,16 @@
...
 	"os/exec"
+
+	"github.com/justwatchcom/gopass/utils/ctxutil"
 )
 
 // Notify displays a desktop notification using osascript
-func Notify(subj, msg string) error {
-	if nn := os.Getenv("GOPASS_NO_NOTIFY"); nn != "" {
+func Notify(ctx context.Context, subj, msg string) error {
+	if os.Getenv("GOPASS_NO_NOTIFY") != "" || ctxutil.IsNotifications(ctx) != true {
```
Comment (dominikschulz) [inline utils/notify/notify_darwin.go]: You should not compare to a bool constant, better: `!ctxutil.IsNotifications(ctx)`

File: utils/notify/notify_linux.go [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
 
 	"github.com/godbus/dbus"
+	"github.com/justwatchcom/gopass/utils/ctxutil"
 )
 
 // Notify displays a desktop notification with dbus
-func Notify(subj, msg string) error {
-	if nn := os.Getenv("GOPASS_NO_NOTIFY"); nn != "" {
+func Notify(ctx context.Context, subj, msg string) error {
+	if os.Getenv("GOPASS_NO_NOTIFY") != "" || ctxutil.IsNotifications(ctx) != true {
```
Comment (dominikschulz) [inline utils/notify/notify_linux.go]: You should not compare to a bool constant, better: `!ctxutil.IsNotifications(ctx)`

File: utils/notify/notify_windows.go [hunk 4]
```diff
@@ -3,13 +3,16 @@
...
 	"os/exec"
+
+	"github.com/justwatchcom/gopass/utils/ctxutil"
 )
 
 // Notify displays a desktop notification through msg
-func Notify(subj, msg string) error {
-	if nn := os.Getenv("GOPASS_NO_NOTIFY"); nn != "" {
+func Notify(ctx context.Context, subj, msg string) error {
+	if os.Getenv("GOPASS_NO_NOTIFY") != "" || ctxutil.IsNotifications(ctx) {
```
Comment (dominikschulz) [inline utils/notify/notify_windows.go]: Missing a negation? This would make more sense: `!ctxutil.IsNotifications(ctx)`

**Comment by ouroboros8** [comment]
😄

**Pull request merged**